src/services/logging.ts

```diff
@@ -0,0 +1,36 @@
+import http from './http';
+
+interface UserAgentDataBrand {
+  readonly brand: string;
+  readonly version: string;
+}
+interface UserAgentData {
+  readonly platform: string;
+  readonly brands: UserAgentDataBrand[];
+  readonly mobile: boolean;
+}
+
+declare global {
+  interface Navigator {
+    readonly userAgentData?: UserAgentData;
+  }
+}
+
+const post = (message: string) => http.post('log', message);
+
+// @TODO: Make this a strcutured component of all logs, e.g. with new columns for OS, Version, etc.
+const parseUserAgentData = () => {
+  if (navigator.userAgentData) {
+    const { brands, platform, mobile } = navigator.userAgentData;
+    const { brand, version } = brands[0];
+    return `${platform} running ${brand} version ${version} on ${mobile ? 'mobile' : 'non-mobile'}`;
+  }
+  return navigator.userAgent;
+};
```

Comment:
I think it would make sense for user agent data to be tracked for every log entry, do you think?

If so, I would rather have this be called inside of the `post` function rather than before it.

Reply:
I agree that it would be great to get it inside all logs (see my comment). I suppose we could just put this at the front of every log message, which is a good enough solution for now. The advantage of using the new API is that this info can't really identify a unique user, so we don't necessarily need to worry about tracking concerns. 

I'll make that change.